cmds/exp/pox/pox.go

```diff
@@ -43,93 +83,134 @@ import (
+	data   string
+	perm   os.FileMode // for target in the chroot
+}{
+	// mount --bind /sys /chroot/sys
+	{"/sys", "/sys", "", syscall.MS_BIND, "", 0555},
+	// mount -t proc /proc /chroot/proc
+	{"/proc", "/proc", "proc", 0, "", 0555},
+	// mount --bind /dev /chroot/dev
+	{"/dev", "/dev", "", syscall.MS_BIND, "", 0755},
+}
+
+func poxCreate(names []string) error {
 	if len(names) == 0 {
 		return fmt.Errorf(usage)
 	}
-
-	if *create {
-		l, err := ldd.Ldd(names)
-		if err != nil {
-			return fmt.Errorf("Running ldd on %v: %v", names, err)
+	l, err := ldd.Ldd(names)
+	if err != nil {
+		var stderr []byte
+		if eerr, ok := err.(*exec.ExitError); ok {
+			stderr = eerr.Stderr
 		}
+		return fmt.Errorf("Running ldd on %v: %v %s", names,
+			err, stderr)
+	}
 
-		for _, dep := range l {
-			v("%s", dep.FullName)
-			names = append(names, dep.FullName)
+	for _, dep := range l {
+		v("%s", dep.FullName)
+		names = append(names, dep.FullName)
+	}
+	// Now we need to make a template file hierarchy and put
+	// the stuff we want in there.
+	dir, err := ioutil.TempDir("", "pox")
+	if err != nil {
+		return err
+	}
+	if !*debug {
+		defer os.RemoveAll(dir)
+	}
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	// We don't use defer() here to close files as
+	// that can cause open failures with a large enough number.
+	for _, f := range names {
+		v("Process %v", f)
+		fi, err := os.Stat(f)
+		if err != nil {
+			return err
 		}
-		// Now we need to make a template file hierarchy and put
-		// the stuff we want in there.
-		dir, err := ioutil.TempDir("", "pox")
+		in, err := os.Open(f)
 		if err != nil {
 			return err
 		}
-		if !*debug {
-			defer os.RemoveAll(dir)
+		f = strings.TrimPrefix(f, pwd)
+		dfile := filepath.Join(dir, f)
+		d := filepath.Dir(dfile)
+		if err := os.MkdirAll(d, 0755); err != nil {
+			in.Close()
+			return err
 		}
-		// We don't use defer() here to close files as
-		// that can cause open failures with a large enough number.
-		for _, f := range names {
-			v("Process %v", f)
-			fi, err := os.Stat(f)
-			if err != nil {
-				return err
-			}
-			in, err := os.Open(f)
-			if err != nil {
-				return err
-			}
-			dfile := filepath.Join(dir, f)
-			d := filepath.Dir(dfile)
-			if err := os.MkdirAll(d, 0755); err != nil {
-				in.Close()
-				return err
-			}
-			out, err := os.OpenFile(dfile, os.O_WRONLY|os.O_CREATE, fi.Mode().Perm())
-			if err != nil {
-				in.Close()
-				return err
-			}
-			_, err = io.Copy(out, in)
+		out, err := os.OpenFile(dfile, os.O_WRONLY|os.O_CREATE,
+			fi.Mode().Perm())
+		if err != nil {
 			in.Close()
-			out.Close()
-			if err != nil {
-				return err
-			}
+			return err
+		}
+		_, err = io.Copy(out, in)
+		in.Close()
+		out.Close()
+		if err != nil {
+			return err
+		}
 
+	}
+	for _, m := range chrootMounts {
+		if err := os.MkdirAll(dir+m.target, m.perm); err != nil {
+			return err
+		}
+	}
+	os.Remove(*file)
+	if *zip {
+		c := exec.Command("zip", "-r", *file, ".")
```

Comment:
pkg/uzip?

Reply:
is that in u-root somewhere?  didn't see it in the pkg directory.  i did see pkg/gzip, but i didn't want to compress a path in place (though i just skimmed it).  i also saw https://golangcode.com/create-zip-files-in-go/, though since we were already calling out to squashfs commands (and i didn't want to remove that), i didn't want to reimplement "recursively zip this entire directory".

Reply:
Ah, sorry, pkg/uzip is in another unmerged PR.